File: backend/app/models/user_profile.py
import uuid

from sqlalchemy import ForeignKey, String, Text
from sqlalchemy.dialects.postgresql import UUID
from sqlalchemy.orm import Mapped, mapped_column, relationship

from app.database.base import Base, TimestampMixin, UUIDMixin


class UserProfile(UUIDMixin, TimestampMixin, Base):
    __tablename__ = "user_profiles"

    user_id: Mapped[uuid.UUID] = mapped_column(
        UUID(as_uuid=True),
        ForeignKey("users.id", ondelete="CASCADE"),
        unique=True,
        nullable=False,
        index=True,
    )
    display_name: Mapped[str] = mapped_column(String(100), nullable=False)
    avatar_url: Mapped[str | None] = mapped_column(String(500), nullable=True)
    bio: Mapped[str | None] = mapped_column(Text, nullable=True)
    timezone: Mapped[str] = mapped_column(String(100), default="UTC", nullable=False)
    # "dark" | "light" | "system"
    theme_preference: Mapped[str] = mapped_column(String(20), default="system", nullable=False)

    user: Mapped["User"] = relationship("User", back_populates="profile")  # type: ignore[name-defined]

    def __repr__(self) -> str:
        return f"<UserProfile user_id={self.user_id} display_name={self.display_name!r}>"

File: mobile/lib/core/models/user_profile.dart
class UserProfileModel {
  final String id;
  final String userId;
  final String displayName;
  final String? avatarUrl;
  final String? bio;
  final String timezone;
  final String themePreference;

  const UserProfileModel({
    required this.id,
    required this.userId,
    required this.displayName,
    this.avatarUrl,
    this.bio,
    required this.timezone,
    required this.themePreference,
  });

  factory UserProfileModel.fromJson(Map<String, dynamic> json) => UserProfileModel(
        id: json['id'] as String,
        userId: json['user_id'] as String,
        displayName: json['display_name'] as String,
        avatarUrl: json['avatar_url'] as String?,
        bio: json['bio'] as String?,
        timezone: json['timezone'] as String? ?? 'UTC',
        themePreference: json['theme_preference'] as String? ?? 'system',
      );
}

class MeResponse {
  final String id;
  final String email;
  final bool isActive;
  final bool isVerified;
  final String createdAt;
  final UserProfileModel? profile;

  const MeResponse({
    required this.id,
    required this.email,
    required this.isActive,
    required this.isVerified,
    required this.createdAt,
    this.profile,
  });

  factory MeResponse.fromJson(Map<String, dynamic> json) => MeResponse(
        id: json['id'] as String,
        email: json['email'] as String,
        isActive: json['is_active'] as bool? ?? true,
        isVerified: json['is_verified'] as bool? ?? false,
        createdAt: json['created_at'] as String,
        profile: json['profile'] == null
            ? null
            : UserProfileModel.fromJson(json['profile'] as Map<String, dynamic>),
      );
}
